ccl/changefeedccl: TestChangefeedJobUpdateFailsIfNotClaimed failed

ccl/changefeedccl.TestChangefeedJobUpdateFailsIfNotClaimed [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9596210?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9596210?buildTab=artifacts#/) on release-23.1 @ [721ace114b1b490e21a2707b578c702293ef2080](https://github.com/cockroachdb/cockroach/commits/721ace114b1b490e21a2707b578c702293ef2080):

Fatal error:

```
panic: test timed out after 59m55s

```
Stack: 

```
goroutine 46369 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2036 +0x88
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x38
```
<details><summary>Log preceding fatal error</summary>
<p>

```
* github.com/cockroachdb/cockroach/pkg/util/admission.initWorkQueue.func2()
* 	github.com/cockroachdb/cockroach/pkg/util/admission/work_queue.go:388 +0x7c
* created by github.com/cockroachdb/cockroach/pkg/util/admission.initWorkQueue
* 	github.com/cockroachdb/cockroach/pkg/util/admission/work_queue.go:385 +0x30c
* 
* goroutine 43286 [chan receive, 2 minutes]:
* github.com/cockroachdb/pebble.(*tableCacheShard).releaseLoop.func1({0x69ea4e8, 0x400bd2c1b0})
* 	github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/table_cache.go:324 +0x80
* runtime/pprof.Do({0x69ea478?, 0x400014e000?}, {{0x40000fd140?, 0x0?, 0x0?}}, 0x4002a8a798)
* 	GOROOT/src/runtime/pprof/runtime.go:40 +0x80
* github.com/cockroachdb/pebble.(*tableCacheShard).releaseLoop(0x0?)
* 	github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/table_cache.go:322 +0x6c
* created by github.com/cockroachdb/pebble.(*tableCacheShard).init
* 	github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/table_cache.go:314 +0xf0
* 
* goroutine 43205 [sleep]:
* time.Sleep(0x4c4b40)
* 	GOROOT/src/runtime/time.go:195 +0x120
* github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Quiesce(0x400a9e0080, {0x69ea478?, 0x400014e000})
* 	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:584 +0x190
* github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Stop(0x400a9e0080, {0x69ea478, 0x400014e000})
* 	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:526 +0xe4
* github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.startTestFullServer.func1()
* 	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/helpers_test.go:410 +0x54
* github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.makeSystemServerWithOptions.func1()
* 	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/helpers_test.go:775 +0x24
* github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.cdcTestNamedWithSystem.func1(0x40046909c0?)
* 	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/helpers_test.go:991 +0x224
* testing.tRunner(0x40046909c0, 0x400bbfb0b0)
* 	GOROOT/src/testing/testing.go:1446 +0x10c
* created by testing.(*T).Run
* 	GOROOT/src/testing/testing.go:1493 +0x328
* 
* goroutine 43278 [chan receive, 2 minutes]:
* github.com/cockroachdb/pebble.(*tableCacheShard).releaseLoop.func1({0x69ea4e8, 0x4009421e30})
* 	github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/table_cache.go:324 +0x80
* runtime/pprof.Do({0x69ea478?, 0x400014e000?}, {{0x40000fd140?, 0x0?, 0x4001a8e808?}}, 0x4001a8e798)
* 	GOROOT/src/runtime/pprof/runtime.go:40 +0x80
* github.com/cockroachdb/pebble.(*tableCacheShard).releaseLoop(0x15c4bdf7061c50d8?)
* 	github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/table_cache.go:322 +0x6c
* created by github.com/cockroachdb/pebble.(*tableCacheShard).init
* 	github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/table_cache.go:314 +0xf0
* 
* goroutine 43272 [select, 2 minutes]:
* github.com/cockroachdb/pebble/vfs.(*diskHealthCheckingFS).startTickerLocked.func1()
* 	github.com/cockroachdb/pebble/vfs/external/com_github_cockroachdb_pebble/vfs/disk_health.go:474 +0xbc
* created by github.com/cockroachdb/pebble/vfs.(*diskHealthCheckingFS).startTickerLocked
* 	github.com/cockroachdb/pebble/vfs/external/com_github_cockroachdb_pebble/vfs/disk_health.go:468 +0x84
* 
*
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/cdc
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedJobUpdateFailsIfNotClaimed.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-26977

--- a/pkg/ccl/changefeedccl/changefeed_test.go
+++ b/pkg/ccl/changefeedccl/changefeed_test.go
@@ -4196,7 +4196,10 @@ func TestChangefeedJobUpdateFailsIfNotClaimed(t *testing.T) {
 		knobs := s.TestingKnobs.DistSQL.(*execinfra.TestingKnobs).Changefeed.(*TestingKnobs)
 		errChan := make(chan error, 1)
 		knobs.HandleDistChangefeedError = func(err error) error {
-			errChan <- err
+			select {
+			case errChan <- err:
+			default:
+			}
 			return err
 		}
 

